mlflow/protos/service.proto

```diff
@@ -707,6 +719,41 @@ message ListExperiments {
   }
 }
 
+message SearchExperiments {
+  option (scalapb.message).extends = "com.databricks.rpc.RPC[$this.Response]";
+
+  // Maximum number of experiments desired.
+  // Servers may select a desired default `max_results` value. All servers are
+  // guaranteed to support a `max_results` threshold of at least 1,000 but may
+  // support more. Callers of this endpoint are encouraged to pass max_results
+  // explicitly and leverage page_token to iterate through experiments.
+  optional int64 max_results = 1;
+
+  // Token indicating the page of experiments to fetch
+  optional string page_token = 2;
+
+  // String representing a SQL filter condition (e.g. "name ILIKE 'my-experiment%'")
+  optional string filter = 3;
+
+  // List of columns for ordering search results, which can include experiment name and id
+  // with an optional "DESC" or "ASC" annotation, where "ASC" is the default.
+  // Tiebreaks are done by experiment id DESC.
+  repeated string order_by = 4;
```

Comment:
On Databricks, we can sort experiments by the following fields:

- ID
- Name
- Basename
- CreationTime (unavailable in OSS)
- LastUpdateTime (unavailable in OSS)

Ref: The experiments table definition in PostgreSQL:

https://github.com/mlflow/mlflow/blob/79d9dfcc68fb7e0441f1acb59901b4d370c0c728/tests/db/schemas/postgresql.sql#L8-L16


Do we need to update the comment here to indicate that supported order-by fields depend on the backend implementation?

Reply:
Or can we just mention what OSS MLflow expects (= experiment name and ID)?

Reply:
I think in we should explicitly say which we can sort by experiment_id and name in doc.

Reply:
Is there a `basename` field in OSS experiment table ?

Reply:
> Is there a basename field in OSS experiment table ?

No